setuptools/command/install.py

```diff
@@ -29,7 +32,8 @@ class install(orig.install):
         'old-and-unmanageable',
         'single-version-externally-managed',
     ]
-    new_commands = [
+    # Type the same as distutils.command.install.install.sub_commands
+    new_commands: ClassVar[list[tuple[str, Callable[[Any], bool] | None]]] = [
```

Comment:
This was causing invariance issues when trying to append `install.new_commands` to `install.sub_commands` at the end of this module